### DISCUSSION TITLE
Solución variables de ENTORNO para OAUTH

### DISCUSSION THREAD

**Discussion by perealberto** [perealberto/decide-single-zambrano#48] [Show and tell]
Equipo, teníamos un problema con las variables de entorno. Al subir un nuevo cambio ocurría lo siguiente:

![image](https://github.com/3ralon/decide-single-zambrano/assets/128732971/cdae060e-7ca8-4080-a065-ce58158b5032)

Es decir, en las github actions se ejecutaba la aplicación. Sin embargo, no era capaz de encontrar la variable de entorno SITE_ID que utilizamos para la feature de Google Oauth porque tenemos configurado en el .gitignore que no se suban los archivos con extensión **.env**.

He estado con @laurolmer viendo posibles soluciones. Finalmente, dimos con la siguiente opción:
- Crear dos archivos de entorno:
  - _.env.personal_: en este tendremos el SITE_ID para nuestra configuración personal de entorno.
  - _.env.github_: en este tendremos un SITE_ID predeterminado para cuando subamos los cambios a github.

![image](https://github.com/3ralon/decide-single-zambrano/assets/128732971/b8bdf1f2-ffec-43d7-a06b-746e75f5cf8e)

De esta manera, los contenedores podrán lanzar la aplicación sin problemas al encontrar la variable de entorno SITE_ID. El código que decide qué archivo de entorno utilizar está en el _settings.py_ de 'decide/decide/':

![image](https://github.com/3ralon/decide-single-zambrano/assets/128732971/ac6bd68a-8677-49f6-a951-a7859c1748ad)

Comentadme qué pensáis @laurolmer @claugp06 @silvglad @MarioArocaPaez

**Comment by claugp06** [comment]
Pues por mí lo veo bien pero ¿Qué site_id se pondría en el de gitHub? porque debería ser el mismo para todos no?
Y una idea aparte una vez que comprobemos que funcione a lo mejor podemos personalizar un  poco más como se hace el relase com por ejemplo que antes del M3 se haga un release de manera automática o algo así.

**Comment by MarioArocaPaez** [comment]
Alberto yo creo que es que no entiende lo que es el var_name quizás con llamalo site_id o algo así si funciona
